### PR TITLE
improve docstrings

### PR DESCRIPTION
**Pull request by CunliangGeng**
Quite many small changes to the docstrings and docs to e.g.
- add code examples
- add link to specific function/class
- add more explanation 
- display API doc better
...

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#266** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/266?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#264** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/264?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/nplinker/arranger.py [hunk 1]
```diff
@@ -27,17 +27,23 @@
 
 
 class DatasetArranger:
-    """Arrange the dataset required by NPLinker.
+    """Arrange datasets based on the fixed working directory structure with the given configuration.
 
-    This class is used to arrange the datasets required by NPLinker according to the
-    configuration. The datasets include MIBiG, GNPS, antiSMASH, and BiG-SCAPE.
+    ??? info "Concept and Diagram"
```
Comment (gcroci2) [inline src/nplinker/arranger.py:32]: Is this a syntax that mkdocs recognizes and links to "Concept and Diagram"?

File: src/nplinker/loader.py [hunk 2]
```diff
@@ -54,12 +60,22 @@ class DatasetLoader:
...
+            config: A Dynaconf object that contains the configuration settings.
+
+        Examples:
+            >>> from nplinker.config import load_config
+            >>> from nplinker.loader import DatasetLoader
+            >>> config = load_config("nplinker.toml")
+            >>> loader = DatasetLoader(config)
+            >>> loader.load()
+
+        See Also:
```
Comment (gcroci2) [inline src/nplinker/loader.py:76]:
```suggestion
        See also:
```

**Review by gcroci2 — APPROVED** [review]

**Comment by gcroci2** [comment]
Before merging, fix the trailing whitespace that makes format checking fail (`src/nplinker/schemas/__init__.py:61:79: W291 Trailing whitespace`) @CunliangGeng

File: src/nplinker/arranger.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/arranger.py:32]: No, that syntax is not for link but for creating a [info block (or callout)](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage), the effect looks like ([you could see that on doc website](https://nplinker.github.io/nplinker/latest/api/arranger/))
image

File: src/nplinker/loader.py (hunk 2 above)
Comment (CunliangGeng) [reply src/nplinker/loader.py:76]: Ruff requires that all section headers in a docstring should be [capitalized](https://docs.astral.sh/ruff/rules/capitalize-section-name/) and will automatically change `also` to `Also`. So let's use `See Also:`.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Jul 12, 4:50 AM EDT**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/266).
* **Jul 12, 4:50 AM EDT**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/266).

**Pull request merged**